File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/TypeTemplateService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbItemCat;
import com.pinyougou.pojo.TbTypeTemplate;

import java.util.List;
import java.util.Map;

public interface TypeTemplateService {
    /** 
    * @Description: 查找到所有的模板信息
    * @Param: [] 
    * @return: java.util.List<com.pinyougou.pojo.TbTypeTemplate> 
    * @Author: Yin 
    * @Date: 2018/9/3 
    */ 
    List<TbTypeTemplate> findAll();

    /** 
    * @Description: 分页查询也就是条件查询 
    * @Param: [page, rows, typeTemplate] 
    * @return: java.util.Map<java.lang.String,java.lang.Object> 
    * @Author: Yin 
    * @Date: 2018/9/3 
    */ 
    Map<String,Object> findByCondition(int page, int rows, TbTypeTemplate typeTemplate);

    /** 
    * @Description: 增加模板的操作 
    * @Param: [typeTemplate] 
    * @return: void 
    * @Author: Yin 
    * @Date: 2018/9/3 
    */ 
    void addTypeTemplate(TbTypeTemplate typeTemplate);

    /** 
    * @Description: 修改模板的操作
    * @Param: [typeTemplate] 
    * @return: void 
    * @Author: Yin 
    * @Date: 2018/9/3 
    */ 
    void updateTypeTemplate(TbTypeTemplate typeTemplate);

    /** 
    * @Description: 根据id查找到模板的操作
    * @Param: [id] 
    * @return: com.pinyougou.pojo.TbTypeTemplate 
    * @Author: Yin 
    * @Date: 2018/9/3 
    */ 
    TbTypeTemplate findById(Long id);

    /** 
    * @Description: 批量删除模板 
    * @Param: [ids] 
    * @return: void 
    * @Author: Yin 
    * @Date: 2018/9/3 
    */ 
    void deleteTypeTemplate(Long[] ids);

    /** 
    * @Description: 为了下拉列表做准备的，也就是进行显示的效果操作
    * @Param: [] 
    * @return: java.util.List<com.pinyougou.pojo.TbItemCat> 
    * @Author: Yin 
    * @Date: 2018/9/7 
    */
    List<Map> findTypeList();

    /** 
    * @Description: 查找到名字 
    * @Param: [id] 
    * @return: com.pinyougou.pojo.TbTypeTemplate 
    * @Author: Yin 
    * @Date: 2018/9/8 
    */ 
    TbTypeTemplate findTypeTemplate(Long id);
}

File: pinyougou-manager-web/src/main/webapp/js/service/typeTemplateService.js
//这是服务层
app.service('typeTemplateService',function ($http) {
    //查找所有
    this.findAll=function () {
        return $http.get("../typeTemplate/findAll.do");
    }

    //条件查询的操作
    this.findByCondition=function (page,rows,entity) {
        return $http.post("../typeTemplate/findByCondition.do?page="+page+"&rows="+rows,entity);
    }

    //增加的操作
    this.add=function (entity) {
        var methodName="add";
        if(entity.id!=null){
            methodName="update";
        }
        return $http.post("../typeTemplate/"+methodName+".do",entity);
    }

    //根据id去找到模板的操作
    this.findOne=function (id) {
        return $http.get("../typeTemplate/findByOne.do?id="+id);
    }

    //删除模板的操作
    this.dele=function (ids) {
        return $http.get("../typeTemplate/delete.do?ids="+ids);
    }

    //下拉列表框的操作，也就是找到名字和id
    this.findTypeList=function () {
        return $http.get("../typeTemplate/findTypeList.do");
    }
})